DOC: How to customize Systems and support new Targets

GregMefford opened the pull request:
I've drafted some docs based on a conversation in Slack about how someone might go about adding support for new Target hardware.

@fhunleth and @mobileoverlord: I haven't been through this process myself yet, so I'm hoping for some feedback about some of the missing details when you have a chance.

fhunleth commented on docs/Systems.md:
```diff
@@ -137,6 +137,60 @@ Would you like to download the system to your cache? [Yn] Y
+1.  Check out their full buildroot and procedure and get their stuff to boot.
+
+2.  Clone buildroot from their main repo and checkout the tag nerves is currently at:
+    https://github.com/nerves-project/nerves_system_br/blob/2d6f76406813734377bc9dad4dd5017e81ffa129/create-build.sh#L16
```
We should refer to `master` rather than the hash tag here. That makes the line reference brittle, but they may be best resolved by pointing the reader to where the Buildroot version is stored in the file.

fhunleth commented on docs/Systems.md:
```diff
@@ -137,6 +137,60 @@ Would you like to download the system to your cache? [Yn] Y
+
+4. Compare the differences between the two to determine what standard things they changed.
+
+  * Anything added like packages and board configs can be copied into `nerves_system_br`.
```
Packages and board configs should be in the system image. The LinkIt Smart repo is an example: https://github.com/nerves-project/nerves_system_linkit

fhunleth commented on docs/Systems.md:
```diff
@@ -137,6 +137,60 @@ Would you like to download the system to your cache? [Yn] Y
+  * Anything added like packages and board configs can be copied into `nerves_system_br`.
+  * Look for patches to existing packages that are needed.
+
+5. Branch buildroot and bring in the relevant changes.
```
I'd like to get rid of the patching Buildroot instructions. It may be necessary for some really tricky environments, but I'd like to discourage it.

GregMefford commented:
@fhunleth: I think I addressed your comments on the previous draft. Let me know if you see other issues or gaps. I'm not convinced there's enough detail yet for someone to do it on their own, but it might be enough that the people wanting to work on CHIP support could get started and ask questions, which we could then clarify in the docs.

GregMefford commented:
@fhunleth Could you clarify what you'd like to see as far as `make system`? Unfortunately, I am not familiar enough with the process to know the details of what's needed, but I'd really like to capture as much as we can in the docs so people can try it and help refine.

mobileoverlord commented:
@GregMefford this has been sitting for a while in limbo, can you clean this up and or close / open a new? Thanks.

GregMefford commented:
I still think there's value in merging this PR with the more specific details @PressY4EKG contributed in #28. I'll work on integrating them so we can at least get this merged with or without a few gaps in it that can be improved later.

GregMefford commented:
I just went through the current docs about [creating or modifying a Nerves System](https://hexdocs.pm/nerves/systems.html#creating-or-modifying-a-nerves-system-with-buildroot) and it seems like what's there is better and more current than what I have in this PR for modifying an existing System, but lacks any discussion about supporting a new Target.

I'm going to make a pass at clarifying a few things based on going through the docs to add a new WiFi driver to an existing system, but I'm not sure how to handle the case of supporting a new platform. I feel like there should at least be something in there warning people that's it's not easy to do, if nothing else.

GregMefford commented:
Alright, I think this PR is ready to be merged now unless you have other feedback or tweaks you'd like to make. Sorry it took me so long to make it happen!

fhunleth commented on docs/Systems.md:
````diff
@@ -220,35 +221,84 @@ cd rpi2_out
+
+```
+NERVES_SYSTEM_CACHE=none
+NERVES_SYSTEM_COMPILER=local
````
I never do this step. I only update `NERVES_SYSTEM`. @mobileoverlord - am I doing it right?

fhunleth commented on docs/Systems.md:
```diff
@@ -220,35 +221,84 @@ cd rpi2_out
-     configuration will need to be updated to point to the custom config.
+> NOTE: You probably want to disable any userland packages that may be included by default to avoid distraction.
 
-The Buildroot [user manual](http://nightly.buildroot.org/manual.html) can be
```
Is the Buildroot documentation linked elsewhere? Given how often I refer to it, it seems important to me to include it somewhere, but I could be talked out of it.

fhunleth replied:
Cool - I see it now.

GregMefford commented:
@mobileoverlord Did you see @fhunleth's question about whether we actually want to have people set the `NERVES_SYSTEM_CACHE=none` and `NERVES_SYSTEM_COMPILER=local` options? I'm not sure whether they're really required in most cases, but I had some notes from a long time ago saying to set them, and I figured it doesn't hurt to make sure you're building locally during custom System development. Let us know if you have feelings about the "right" or "best" way we should be telling people to do it.
